lib/bankaccounts/checking-account.rb

```diff
@@ -0,0 +1,48 @@
+module BankAccounts
+
+  # CheckingAccount inherits from Account class
+  class CheckingAccount < Account
+    attr_accessor :checks_used
+
+    def initialize(id, initial_balance)
+      super
+      @min_balance = 0
+      check_initial_balance(initial_balance)
+      @fee = 1
+      @checks_used = 0
+    end
+
+    # withdraw is inherited from Account; therefore, it does not appear here.
+    # withdraw(amount) subtracts amount from balance, incurs $1 fee.
+    # if account would fall below @min_balance, outputs warning & returns unmodified balance.
+
+    def withdraw_using_check(amount)
+      # re-assigns @min_balance to allow overdraft up to -$10
+      @min_balance = -10
```

Comment:
Ah, be careful. You're reassigning both `@min_balance` and `@fee`, so now if someone uses `#withdraw_using_check` first, then `#withdraw`, you'd end up with incorrect conditionals happening.

What you probably want to do is use local variables that exist only in the context of `#withdraw_using_check`, so it doesn't affect `#withdraw`